src/utils/__tests__/internalLinks-test.js

```diff
@@ -37,6 +37,100 @@ describe('isInternalLink', () => {
     [false, 'same domain, double slash', 'https://example.com//#narrow/stream/jest'],
     [false, 'same domain, triple slash', 'https://example.com///#narrow/stream/jest'],
 
+    // These examples may be odd-looking URLs, but are nevertheless valid.
+    [true, 'scheme-relative', '//example.com/#narrow/stream/jest'],
+    [true, 'path-relative, nop path', '.#narrow/stream/jest'],
+    [true, 'path-relative, nop path', './#narrow/stream/jest'],
+    [true, 'path-relative, nop path', '..#narrow/stream/jest'],
+    [true, 'path-relative, nop path', '../#narrow/stream/jest'],
+    [true, 'path-relative, nop path', 'foo/..#narrow/stream/jest'],
+    [true, 'path-relative, nop path', 'foo/../#narrow/stream/jest'],
+    [true, 'path-relative, nop path', 'foo/bar/../../../baz/.././#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/.#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/./#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/..#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/../#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/foo/..#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/foo/../#narrow/stream/jest'],
+    [true, 'path-absolute, nop path', '/foo/bar/../../../baz/.././#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/.#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/./#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/..#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/../#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/foo/..#narrow/stream/jest'],
+    [true, 'same domain, nop path', 'https://example.com/foo/../#narrow/stream/jest'],
+    [
+      true,
+      'same domain, nop path',
+      'https://example.com/foo/bar/../../../baz/.././#narrow/stream/jest',
+    ],
+    [true, 'same domain, %-encoded host', 'https://%65xample%2ecom/#narrow/stream/jest'],
+    [
+      true,
+      'same domain, punycoded host',
+      'https://example.xn--h2brj9c/#narrow/stream/jest',
+      new URL('https://example.भारत/'),
+    ],
+    [
+      true,
+      'same domain, punycodable host',
+      'https://example.भारत/#narrow/stream/jest',
+      new URL('https://example.भारत/'),
+    ],
+    // This one fails because our polyfilled URL implementation has IDNA stripped out.
+    // [true, 'same domain, IDNA-mappable', 'https://ℯⅩªm🄿ₗℰ.ℭᴼⓂ/'], // FAILS
+    [
+      true,
+      'same IPv4 address, %-encoded',
+      'http://%31%39%32%2e168%2e0%2e1/#narrow/stream/jest',
+      new URL('http://192.168.0.1/'),
+    ],
+    // This one fails because our polyfilled URL implementation has IDNA stripped out.
+    // [true, 'same IPv4 address, IDNA-mappable', 'http://１𝟗𝟚。①⁶🯸．₀｡𝟭/#narrow/stream/jest', new URL('http://192.168.0.1/'),],
```

Comment:
Same comment as above, about using the polyfill.

This one passes when I `nvm use 13` or `nvm use 14`, but not when I `nvm use 12`; that points to the polyfill not being used here.

Reply:
Hmm interesting! That sure seems informative. I'll have to investigate a bit more.

Reply:
Yeah, I see. We get our polyfilled `URL` implementation in the app because of this line by the top of `src/ZulipMobile.js`:
```js
import 'react-native-url-polyfill/auto';
```
But none of our tests import that file, and we don't do anything else to apply the polyfill, so we don't get it in tests.

If I add that polyfill line at the top of the test file:
```
import 'react-native-url-polyfill/auto'
```
then the test fails even with Node 14. So does another one, which in fact had been kind of puzzling before!
```js
    [
      true,
      'same domain, punycoded host',
      'https://example.xn--h2brj9c/#narrow/stream/jest',
      new URL('https://example.भारत/'),
    ],
```
I guess that puzzlingness was a sign telling me to look deeper.

Seems like the right thing to do here is that we should be applying the same polyfill in tests as we do in the app -- otherwise we're at high risk of confusing ourselves anywhere that the polyfill and the implementation in Node have any different behavior. I'll add a commit to that effect.